### ISSUE TITLE
uid/gid related permissions checking methods for t.w.filepath

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7610]
|<img alt="slyphon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| slyphon reported|
|-|-|
|Trac ID|trac#405|
|Type|enhancement|
|Created|2003-11-19 13:22:43Z|

<details><summary>Searchable metadata</summary>

```
trac-id__405 405
type__enhancement enhancement
reporter__slyphon slyphon
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__invalid invalid
component__core core
keywords__ 
time__1069248163000000 1069248163000000
changetime__1469402313501175 1469402313501175
version__ 
owner__ 
cc__glyph cc__radix cc__spiv cc__itamarst cc__slyphon
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="slyphon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|slyphon commented|
|-|-|

```
#!html
<pre>
as per a discussion last night, i think it would be useful to add some
permissions-checking methods to twisted.web.filepath. I've written one that will
be unix-centric (but effective). These methods would be of great use to FTP. In
fact i was rather upset last night when i learned about filepath, because I'd
already spent a good deal of time re-inventing that wheel...

comments?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

There is no such thing as t.w.filepath any longer.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub commented|
|-|-|

In changeset 09316b0dfbc297d03d88ab2cb14ed5481c11b51f
```
#!CommitTicketReference repository="" revision="09316b0dfbc297d03d88ab2cb14ed5481c11b51f"
Merge pull request #405 from twisted/8662-sshforwardwin

Author: hawkowl
Reviewer: rodrigc
Fixes: #8662
```
